d2layouts/d2dagrelayout/layout.go

```diff
@@ -257,7 +258,11 @@ func setGraphAttrs(attrs dagreGraphAttrs) string {
 }
 
 func escapeID(id string) string {
-	id = strings.ReplaceAll(id, `\n`, `\\n`)
+	// fixes \\
+	id = strings.ReplaceAll(id, "\\", `\\`)
+	// replaces \n with \\n whenever \n is not preceded by \ (does not replace \\n)
+	re := regexp.MustCompile(`[^\\](\n)`)
```

Comment:
Why the paren around `\n`?

Reply:
My bad, for a moment I thought I had to replace the group and ended forgetting it there 😅